adapters/repos/db/lsmkv/contentReader/content_reader.go

```diff
@@ -0,0 +1,248 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package contentReader
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+	"syscall"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+
+	"github.com/edsrzf/mmap-go"
+)
+
+const (
+	uint64Len = 8
+	uint32Len = 4
+)
+
+type ContentReader interface {
+	ReadValue(offset uint64) (byte, uint64)
+	ReadRange(offset uint64, length uint64, outBuf []byte) ([]byte, uint64)
+	ReadUint64(offset uint64) (uint64, uint64)
```

Comment:
We should add endianess as a parameter or additional method, as we mix it in the codebase (e.g. docIds in searchable index are big endian)